### PR TITLE
feat(adapter-xhr): Support `context` option

### LINKED ISSUE #204: Support `context` option on the XHR Adapter
Add support for mocking the XHR global on a given context (like we currently support with the fetch adapter).

```js
{
  adapterOptions: { 
    xhr: { context: win }
  }
}
```

### PR DESCRIPTION
**Pull request by offirgolan**
## Description

Add support for mocking the XHR global on a given context (like we currently support with the fetch adapter).

```js
{
  adapterOptions: { 
    xhr: { context: win }
  }
}
```

## Motivation and Context

Resolves #204.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have added tests to cover my changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] My code follows the code style of this project.
- [x] My commits and the title of this PR follow the [Conventional Commits Specification](https://www.conventionalcommits.org).
- [x] I have read the [contributing guidelines](https://github.com/Netflix/pollyjs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: packages/@pollyjs/adapter-xhr/src/index.js [hunk 1]
```diff
@@ -36,11 +45,13 @@ export default class XHRAdapter extends Adapter {
       };
     };
 
-    global.XMLHttpRequest[IS_STUBBED] = true;
+    context.XMLHttpRequest[IS_STUBBED] = true;
```
Comment (jasonmit) [inline packages/@pollyjs/adapter-xhr/src/index.js]: Weakmap could work here so we do mutate the objects we don't own

**Comment by freewind** [comment]
Really need this feature, hope it get merged early

File: packages/@pollyjs/adapter-xhr/src/index.js (hunk 1 above)
Comment (offirgolan) [reply packages/@pollyjs/adapter-xhr/src/index.js]: Updated to use a `WeakSet`

**Comment by offirgolan** [comment]
Released with `v2.5.0`.

**Pull request merged**